synapse/handlers/sync.py

```diff
@@ -562,8 +562,21 @@ def compute_summary(self, room_id, sync_config, batch, state, now_token):
             details.get(Membership.INVITE, empty_ms).count
         )
 
-        if name_id or canonical_alias_id:
-            defer.returnValue(summary)
+        # if the room has a (non-deleted) name or canonical_alias, we can skip
+        # calculating heroes.  we assume that if the event has contents, it'll
+        # be a valid name or canonical_alias - i.e. we're checking that they
+        # haven't been deleted.
+        if name_id:
+            name = yield self.store.get_event(name_id, allow_none=False)
+            if name.content:
```

Comment:
You also need to check that the returned event is not None

Reply:
oops, thanks